File: objectiveai-sdk-rs/src/agent/http.rs
//! HTTP client functions for Agent endpoints.

use crate::{HttpClient, HttpError};

/// Lists all Agents that have been used.
pub async fn list_agents(
    client: &HttpClient,
    params: super::request::ListAgentsRequest,
) -> Result<super::response::ListAgentResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "agents/list", Some(params))
        .await
}

/// Retrieves a specific Agent.
pub async fn get_agent(
    client: &HttpClient,
    params: super::request::GetAgentRequest,
) -> Result<super::response::GetAgentResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "agents", Some(params))
        .await
}

/// Retrieves usage statistics for a specific Agent.
pub async fn get_agent_usage(
    client: &HttpClient,
    params: super::request::GetAgentRequest,
) -> Result<super::response::UsageAgentResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "agents/usage", Some(params))
        .await
}

File: objectiveai-sdk-py/objectiveai_sdk/agent/http.py
"""HTTP functions for agent endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai_sdk.agent import (
        GetAgentResponse,
        ListAgentResponse,
        ListAgentsRequest,
        GetAgentRequest,
        UsageAgentResponse,
    )
    from objectiveai_sdk.client import ObjectiveAI


async def list_agents(
    client: ObjectiveAI, params: ListAgentsRequest,
) -> ListAgentResponse:
    """List all agents that have been used."""
    return await client.post_unary("agents/list", params)


async def get_agent(
    client: ObjectiveAI, params: GetAgentRequest,
) -> GetAgentResponse:
    """Retrieve a specific agent."""
    return await client.post_unary("agents", params)


async def get_agent_usage(
    client: ObjectiveAI, params: GetAgentRequest,
) -> UsageAgentResponse:
    """Retrieve usage statistics for a specific agent."""
    return await client.post_unary("agents/usage", params)
